src/analysis/SmartDetectorsAnalysis/AutomaticResolutionState.cs

```diff
@@ -0,0 +1,27 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AutomaticResolutionState.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliance.Analysis
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A class encapsulating the automatic resolution state, used to extract all relevant
+    /// alert information for supplying the detector on automatic resolution checks.
+    /// </summary>
+    public class AutomaticResolutionState
+    {
+        /// <summary>
+        /// Gets or sets the alert ID this state relates to.
+        /// </summary>
+        public string AlertId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the alert predicates this state relates to.
+        /// </summary>
+        public IReadOnlyDictionary<string, object> AlertPredicates { get; set; }
```

Comment:
>AlertPredicates [](start = 51, length = 15)

why do we need this?

Reply:
removed

---
In reply to: [235355257](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r235355257) [](ancestors = 235355257)

Reply:
talked it F2F

---
In reply to: [235355257](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r235355257) [](ancestors = 235355257)